src/components/DateInput/DateInput.tsx

```diff
@@ -102,16 +112,20 @@ const DateInput: React.FC<DateInputProps & Omit<TextInputProps, 'value' | 'onCha
     close();
   }, [close, value, timezone, setCurrentDate]);
 
+  const onClear = useCallback(() => setCurrentDate(undefined), [setCurrentDate]);
+
   const onApply = useCallback(
     e => {
-      if (!currentDate) {
+      if (!currentDate && disableReset) {
         return;
       }
       e.preventDefault();
-      onChange(currentDate.startOf(withTime ? 'minute' : 'day').toDate());
+      onChange(
+        !currentDate ? currentDate : currentDate.startOf(withTime ? 'minute' : 'day').toDate()
```

Comment:
out ouf curiosity, since `currentDate` is undefined, wouldn't that work? The ternary would quickly yield `undefined`, no?

```suggestion
        currentDate?.startOf(withTime ? 'minute' : 'day').toDate()
```